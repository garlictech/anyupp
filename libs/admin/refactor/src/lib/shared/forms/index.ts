import { AdminSharedFormsModule } from './admin-shared-forms.module';
import { AbstractFormDialogComponent } from './components/abstract-form-dialog/abstract-form-dialog.component';
import { FormsService } from './services/forms/forms.service';

export { AbstractFormDialogComponent, AdminSharedFormsModule, FormsService };
