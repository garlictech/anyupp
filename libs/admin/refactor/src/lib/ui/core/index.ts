export * from './admin-ui-core.module';
export * from './utils';
