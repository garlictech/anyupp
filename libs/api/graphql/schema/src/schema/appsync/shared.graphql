### COMMON
# TODO: extraxt these because the admin-api also contains these
type Location {
  lat: String!
  lng: String!
}

input LocationInput {
  lat: String!
  lng: String!
}

type Address {
  address: String
  city: String
  country: String
  title: String
  postalCode: String
  location: Location
}

type LocalizedItem {
  en: String
  de: String
  hu: String
}

type PaymentMode {
  name: String!
  caption: String
  method: PaymentMethod!
}
enum PaymentMethod {
  CASH
  CARD
  INAPP
}

type ChainStyle {
  colors: ChainStyleColors!
  images: ChainStyleImages
}

type ChainStyleColors {
  backgroundLight: String!
  backgroundDark: String!
  borderLight: String!
  borderDark: String!
  disabled: String!
  highlight: String!
  indicator: String!
  textLight: String!
  textDark: String!
}

type ChainStyleImages {
  header: String
  logo: String
}
