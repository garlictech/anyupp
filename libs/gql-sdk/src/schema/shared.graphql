### Location
type Location @aws_cognito_user_pools {
  lat: Float!
  lng: Float!
}
input LocationInput {
  lat: Float!
  lng: Float!
}

### LocalizedItem
type LocalizedItem @aws_cognito_user_pools {
  en: String
  de: String
  hu: String
}
input LocalizedItemInput {
  en: String
  de: String
  hu: String
}

type Address @aws_cognito_user_pools {
  address: String!
  city: String!
  country: String!
  title: String!
  postalCode: String!
  location: Location!
}

### Payment
type PaymentMode @aws_cognito_user_pools {
  type: PaymentType!
  caption: String
  method: PaymentMethod!
}

enum PaymentMethod {
  cash
  card
  inapp
}

enum PaymentType {
  card
  cash
  simple
  stripe
  applepay
  googlepay
}

### ChainStyle
type ChainStyle @aws_cognito_user_pools {
  colors: ChainStyleColors!
  images: ChainStyleImages
}

type ChainStyleColors @aws_cognito_user_pools {
  backgroundLight: String!
  backgroundDark: String!
  borderLight: String!
  borderDark: String!
  disabled: String!
  highlight: String!
  indicator: String!
  textLight: String!
  textDark: String!
}

type ChainStyleImages @aws_cognito_user_pools {
  header: String
  logo: String
}

### ChainProduct
type ChainProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  chainId: ID!
  name: LocalizedItem!
  description: LocalizedItem
  productCategoryId: ID!
  productType: String!
  isVisible: Boolean!
  image: String
  variants: [ProductVariant]
  allergens: [Allergen]
  configSets: [ProductConfigSet]
}
input UpdateChainProductInput {
  id: ID!
  chainId: ID
  name: LocalizedItemInput
  description: LocalizedItemInput
  productCategoryId: ID
  productType: String
  isVisible: Boolean
  image: String
  variants: [ProductVariantInput]
  allergens: [Allergen]
  configSets: [ProductConfigSetInput]
}

enum Allergen {
  celery
  crustaceans
  egg
  fish
  gluten
  lupin
  milk
  molluscs
  mustard
  peanut
  sesame
  soya
  sulphites
  treenuts
}

enum ServingMode {
  inPlace
  takeAway
}

enum OrderMode {
  instant
  pickup
}

### GroupProduct
type GroupProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  parentId: ID! # parent chainProduct ID
  chainId: ID!
  groupId: ID!
  isVisible: Boolean!
  tax: Int!
  takeawayTax: Int
  variants: [ProductVariant]
  configSets: [ProductConfigSet]
}
input UpdateGroupProductInput {
  id: ID!
  parentId: ID
  chainId: ID
  groupId: ID
  isVisible: Boolean
  tax: Int
  takeawayTax: Int
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
}

### UnitProduct
type UnitProduct
  @aws_cognito_user_pools
  @model
  @searchable
  @auth(rules: [{ allow: private }, { allow: public, provider: iam }]) {
  id: ID!
  parentId: ID! # parent groupProduct ID
  chainId: ID!
  groupId: ID!
  unitId: ID!
  isVisible: Boolean!
  takeaway: Boolean! @deprecated(reason: "Use `supportedServingModes`.")
  laneId: ID
  position: Int!
  variants: [ProductVariant]
  configSets: [ProductConfigSet]
  supportedServingModes: [ServingMode!]!
}
input CreateUnitProductInput {
  id: ID # <== NOT REQUIRED
  parentId: ID! # parent groupProduct ID
  chainId: ID!
  groupId: ID!
  unitId: ID!
  isVisible: Boolean!
  takeaway: Boolean!
  laneId: ID
  position: Int!
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
  supportedServingModes: [ServingMode!]!
}
input UpdateUnitProductInput {
  id: ID! # <== REQUIRED
  parentId: ID
  chainId: ID
  groupId: ID
  unitId: ID
  isVisible: Boolean
  takeaway: Boolean
  laneId: ID
  position: Int
  variants: [ProductVariantInput]
  configSets: [ProductConfigSetInput]
  supportedServingModes: [ServingMode!]
}

### ProductConfigSet
type ProductConfigComponent @aws_cognito_user_pools {
  productComponentId: ID!
  refGroupPrice: Float!
  price: Float!
  position: Int!
}
input ProductConfigComponentInput {
  productComponentId: ID!
  refGroupPrice: Float!
  price: Float!
  position: Int!
}

type ProductConfigSet @aws_cognito_user_pools {
  productSetId: ID!
  items: [ProductConfigComponent!]!
  position: Int!
}
input ProductConfigSetInput {
  productSetId: ID!
  items: [ProductConfigComponentInput!]!
  position: Int!
}

### ProductVariant
type ProductVariant @aws_cognito_user_pools {
  id: ID!
  variantName: LocalizedItem!
  pack: ProductVariantPack
  refGroupPrice: Float
  isAvailable: Boolean!
  price: Float!
  availabilities: [Availability]
  position: Int!
}
input ProductVariantInput {
  id: ID!
  variantName: LocalizedItemInput!
  pack: ProductVariantPackInput
  refGroupPrice: Float
  isAvailable: Boolean!
  price: Float!
  availabilities: [AvailabilityInput]
  position: Int!
}

### ProductVariantPack
type ProductVariantPack @aws_cognito_user_pools {
  size: Float!
  unit: String!
}
input ProductVariantPackInput {
  size: Float!
  unit: String!
}

### Availability
type Availability @aws_cognito_user_pools {
  type: String!
  dayFrom: String
  dayTo: String
  timeFrom: String
  timeTo: String
  price: Float!
}
input AvailabilityInput {
  type: String!
  dayFrom: String
  dayTo: String
  timeFrom: String
  timeTo: String
  price: Float!
}

### UserInvoiceAddress
# Re enable (issue: #1736)
# type UserInvoiceAddress {
#   customerName: String!
#   taxNumber: String!
#   country: String!
#   city: String!
#   streetAddress: String!
#   postalCode: String!
#   email: String
# }
# input UserInvoiceAddressInput {
#   customerName: String!
#   taxNumber: String!
#   country: String!
#   city: String!
#   streetAddress: String!
#   postalCode: String!
#   email: String
# }
