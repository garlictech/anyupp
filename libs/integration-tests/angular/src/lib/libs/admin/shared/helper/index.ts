import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
