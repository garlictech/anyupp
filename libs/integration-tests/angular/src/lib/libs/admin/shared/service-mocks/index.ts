export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
