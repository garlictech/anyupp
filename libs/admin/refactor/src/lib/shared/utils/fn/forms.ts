import { isNumber, omit } from 'lodash/fp';

import {
  AbstractControl,
  FormBuilder,
  ValidatorFn,
  Validators,
} from '@angular/forms';
import { Address, LocationLatLon } from '@bgap/domain';
import { EVariantAvailabilityType } from '@bgap/shared/types';

import { WEEKLY_VARIANT_AVAILABILITY } from '../const';

export const contactFormGroup = (requiredEmail = false) => ({
  email: requiredEmail
    ? ['', [Validators.email, Validators.required]]
    : ['', [Validators.email]],
  phone: [''],
});

export const addressFormGroup = (
  formBuilder: FormBuilder,
  required = false,
) => ({
  address: formBuilder.group(
    {
      address: ['', required ? [Validators.required] : []],
      city: ['', required ? [Validators.required] : []],
      country: ['', required ? [Validators.required] : []],
      title: [''],
      postalCode: ['', required ? [Validators.required] : []],
    },
    required ? {} : { validators: emptyAddressValidator },
  ),
});

export const locationFormGroup = (
  formBuilder: FormBuilder,
  required = false,
) => ({
  location: formBuilder.group(
    {
      lat: [0],
      lon: [0],
    },
    required ? {} : { validators: emptyLocationValidator },
  ),
});

export const multiLangValidator: ValidatorFn = (control: AbstractControl) => {
  const hu = control.get('hu')?.value;
  const en = control.get('en')?.value;
  const de = control.get('de')?.value;

  return hu || en || de ? null : { empty: true };
};

export const addressIsEmpty = (address?: Address) => {
  const stringFields = omit(['location'], address);
  const allStringsAreEmpty = Object.values(stringFields).every(v => !v);

  return allStringsAreEmpty;
};

export const addressIsFilled = (address: Address) => {
  const stringFields = omit(['location'], address);
  const allStringsAreFilled = Object.values(stringFields).every(v => v);

  return allStringsAreFilled;
};

export const locationIsFilled = (location: LocationLatLon) =>
  isNumber(location?.lat) &&
  isNumber(location?.lon) &&
  (location?.lat !== 0 || location?.lon !== 0);

// Address: all fields are empty, or all fields are required.
export const emptyAddressValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const isEmpty = addressIsEmpty(control.value);
  const isFilled = addressIsFilled(control.value);

  return isEmpty || isFilled ? null : { err: true };
};

// Address: all fields are empty, or all fields are required.
export const emptyLocationValidator: ValidatorFn = (control: AbstractControl) =>
  locationIsFilled(control.value) ? null : { err: true };

const getTimeError = (timeFrom: string, timeTo: string) =>
  timeFrom && timeTo
    ? timeFrom < timeTo
      ? null
      : { timeInterval: true }
    : { missingTime: true };

const getWeeklyDayError = (dayFrom: string, dayTo: string, days: string[]) =>
  dayFrom && dayTo
    ? days.indexOf(dayFrom) <= days.indexOf(dayTo)
      ? null
      : { dayInterval: true }
    : { missingDay: true };

const getSeasonalDayError = (dayFrom: string, dayTo: string) =>
  dayFrom && dayTo
    ? dayFrom <= dayTo
      ? null
      : { dayInterval: true }
    : { missingDay: true };

export const productAvailabilityValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const type = control.get('type')?.value;
  const dayFrom = control.get('dayFrom')?.value;
  const dayTo = control.get('dayTo')?.value;
  const timeFrom = control.get('timeFrom')?.value;
  const timeTo = control.get('timeTo')?.value;

  if (type === EVariantAvailabilityType.WEEKLY) {
    return (
      getWeeklyDayError(
        dayFrom,
        dayTo,
        Object.keys(WEEKLY_VARIANT_AVAILABILITY),
      ) ||
      getTimeError(timeFrom, timeTo) ||
      null
    );
  } else if (type === EVariantAvailabilityType.SEASONAL) {
    return (
      getSeasonalDayError(dayFrom, dayTo) ||
      getTimeError(timeFrom, timeTo) ||
      null
    );
  }

  return null;
};

export const dailyScheduleBothEmptyOrProperlyFilledValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const from = control.get('from')?.value;
  const to = control.get('to')?.value;

  return (!!from && !!to) || (!from && !to)
    ? control.get('from')?.valid && control.get('to')?.valid
      ? null
      : { timeFormat: true }
    : { missingIntervall: true };
};

export const makeId = (length: number): string => {
  let result = '';
  const chars =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
  const charsLength = chars.length;

  for (let i = 0; i < length; i++) {
    result += chars.charAt(Math.floor(Math.random() * charsLength));
  }

  return result;
};

export const optionalValueValidator: ValidatorFn = (control: AbstractControl) =>
  !control.value || !!control.value ? null : { err: true };

export const notEmptyArray: ValidatorFn = (control: AbstractControl) =>
  control.value.length > 0 ? null : { err: true };

export const getInitials = (text: string) => {
  const parts = (text || '').split(' ').filter(i => !!i);

  return (
    (parts[0] || '-').charAt(0) + (parts[1] ? parts[1].charAt(0) : '')
  ).toUpperCase();
};
