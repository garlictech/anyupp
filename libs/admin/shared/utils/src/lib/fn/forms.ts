import { isNumber, omit } from 'lodash/fp';
import { DateTime } from 'luxon';

import {
  AbstractControl,
  FormBuilder,
  ValidatorFn,
  Validators,
} from '@angular/forms';
import * as CrudApi from '@bgap/crud-gql/api';
import { EVariantAvailabilityType, DateIntervals } from '@bgap/shared/types';

import { WEEKLY_VARIANT_AVAILABILITY } from '../const';

export const contactFormGroup = (requiredEmail = false) => ({
  email: requiredEmail
    ? ['', [Validators.email, Validators.required]]
    : ['', [Validators.email]],
  phone: [''],
});

export const addressFormGroup = (
  formBuilder: FormBuilder,
  required = false,
) => ({
  address: formBuilder.group(
    {
      address: ['', required ? [Validators.required] : []],
      city: ['', required ? [Validators.required] : []],
      country: ['', required ? [Validators.required] : []],
      title: [''],
      postalCode: ['', required ? [Validators.required] : []],
      location: formBuilder.group({
        lat: [0],
        lng: [0],
      }),
    },
    required ? {} : { validators: emptyAddressValidator },
  ),
});

export const multiLangValidator: ValidatorFn = (control: AbstractControl) => {
  const hu = control.get('hu')?.value;
  const en = control.get('en')?.value;
  const de = control.get('de')?.value;

  return hu || en || de ? null : { empty: true };
};

export const addressIsEmpty = (address: CrudApi.Address) => {
  const stringFields = omit(['location'], address);
  const allStringsAreEmpty = Object.values(stringFields).every(v => !v);

  // 0 or empty string
  const locationIsEmpty = !address.location?.lat && !address.location?.lng;

  return allStringsAreEmpty && locationIsEmpty;
};

export const addressIsFilled = (address: CrudApi.Address) => {
  const stringFields = omit(['location'], address);
  const allStringsAreFilled = Object.values(stringFields).every(v => v);

  // not 0 numeric
  const locationIsFilled =
    isNumber(address.location?.lat) &&
    isNumber(address.location?.lng) &&
    (address.location?.lat !== 0 || address.location?.lng !== 0);

  return allStringsAreFilled && locationIsFilled;
};

// Address: all fields are empty, or all fields are required.
export const emptyAddressValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const isEmpty = addressIsEmpty(control.value);
  const isFilled = addressIsFilled(control.value);

  return isEmpty || isFilled ? null : { err: true };
};

export const productAvailabilityValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const type = control.get('type')?.value;
  const dayFrom = control.get('dayFrom')?.value;
  const dayTo = control.get('dayTo')?.value;
  const timeFrom = control.get('timeFrom')?.value;
  const timeTo = control.get('timeTo')?.value;

  if (type === EVariantAvailabilityType.WEEKLY) {
    const days = Object.keys(WEEKLY_VARIANT_AVAILABILITY);

    const dayError =
      dayFrom && dayTo
        ? days.indexOf(dayFrom) <= days.indexOf(dayTo)
          ? null
          : { dayInterval: true }
        : { missingDay: true };

    const timeError =
      timeFrom && timeTo
        ? timeFrom < timeTo
          ? null
          : { timeInterval: true }
        : { missingTime: true };

    return dayError || timeError || null;
  } else if (type === EVariantAvailabilityType.SEASONAL) {
    const dayError =
      dayFrom && dayTo
        ? dayFrom <= dayTo
          ? null
          : { dayInterval: true }
        : { missingDay: true };
    const timeError =
      timeFrom && timeTo
        ? timeFrom < timeTo
          ? null
          : { timeInterval: true }
        : { missingTime: true };

    return dayError || timeError || null;
  }

  return null;
};

export const dailyScheduleBothEmptyOrProperlyFilledValidator: ValidatorFn = (
  control: AbstractControl,
) => {
  const from = control.get('from')?.value;
  const to = control.get('to')?.value;

  return (!!from && !!to) || (!from && !to)
    ? control.get('from')?.valid && control.get('to')?.valid
      ? null
      : { timeFormat: true }
    : { missingIntervall: true };
};

export const getDayIntervals = (
  dateValue: string | number,
  timeZone: string,
): DateIntervals => {
  const date: DateTime = DateTime.fromISO(new Date(dateValue).toISOString(), {
    zone: timeZone,
  });

  return {
    from: date.startOf('day').valueOf(),
    to: date.endOf('day').valueOf(),
  };
};

export const makeId = (length: number): string => {
  let result = '';
  const chars =
    'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
  const charsLength = chars.length;

  for (let i = 0; i < length; i++) {
    result += chars.charAt(Math.floor(Math.random() * charsLength));
  }

  return result;
};

export const optionalValueValidator: ValidatorFn = (control: AbstractControl) =>
  !control.value || !!control.value ? null : { err: true };

export const notEmptyArray: ValidatorFn = (control: AbstractControl) =>
  control.value.length > 0 ? null : { err: true };
