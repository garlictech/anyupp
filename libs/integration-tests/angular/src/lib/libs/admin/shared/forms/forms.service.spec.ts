import { TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { FormsService } from '@bgap/admin/shared/forms';
import { StoreModule } from '@ngrx/store';

describe('FormsService', () => {
  let service: FormsService;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, StoreModule.forRoot({})],
    });

    service = TestBed.inject(FormsService);
  });

  it('createProductVariantFormGroup should create form group', () => {
    expect(service.createProductVariantFormGroup().value).toMatchSnapshot({
      id: expect.any(String),
    });
  });

  it('createProductAvailabilityFormGroup should create form group', () => {
    expect(
      service.createProductAvailabilityFormGroup().value,
    ).toMatchSnapshot();
  });

  it('createCustomDailyScheduleFormGroup should create form group', () => {
    expect(
      service.createCustomDailyScheduleFormGroup().value,
    ).toMatchSnapshot();
  });

  it('createLaneFormGroup should create form group', () => {
    expect(service.createLaneFormGroup().value).toMatchSnapshot({
      id: expect.any(String),
    });
  });

  it('createProductConfigSetFormGroup should create form group', () => {
    expect(service.createProductConfigSetFormGroup().value).toMatchSnapshot();
  });

  it('createProductConfigSetItemFormGroup should create form group', () => {
    expect(
      service.createProductConfigSetItemFormGroup().value,
    ).toMatchSnapshot();
  });

  it('createRkeeperFormGroup should create form group with anyupp password', () => {
    expect(service.createRkeeperFormGroup().value).toMatchSnapshot();
  });

  it('createRkeeperFormGroup should create form group without anyupp password', () => {
    expect(service.createRkeeperFormGroup().value).toMatchSnapshot();
  });

  it('createRatingPolicyFormGroup should create form group ', () => {
    expect(service.createRatingPolicyFormGroup().value).toMatchSnapshot();
  });

  it('createTipPolicyFormGroup should create form group ', () => {
    expect(service.createTipPolicyFormGroup().value).toMatchSnapshot();
  });

  it('addUnitTipPercent should add percent value to an array', () => {
    expect(service.addUnitTipPercent('10', [])).toMatchInlineSnapshot(`
      Array [
        10,
      ]
    `);
    expect(service.addUnitTipPercent('30', [10, 20])).toMatchInlineSnapshot(`
      Array [
        10,
        20,
        30,
      ]
    `);
    expect(service.addUnitTipPercent('20', [10, 30])).toMatchInlineSnapshot(`
      Array [
        10,
        20,
        30,
      ]
    `);
  });

  it('removeUnitTipPercent should remove percent value from an array', () => {
    expect(service.removeUnitTipPercent('10', [])).toMatchInlineSnapshot(
      `Array []`,
    );
    expect(service.removeUnitTipPercent('10', [10, 20, 30]))
      .toMatchInlineSnapshot(`
      Array [
        20,
        30,
      ]
    `);
    expect(service.removeUnitTipPercent('30', [10, 20, 30]))
      .toMatchInlineSnapshot(`
      Array [
        10,
        20,
      ]
    `);
    expect(service.removeUnitTipPercent('33', [10, 20])).toMatchInlineSnapshot(`
      Array [
        10,
        20,
      ]
    `);
  });
});
