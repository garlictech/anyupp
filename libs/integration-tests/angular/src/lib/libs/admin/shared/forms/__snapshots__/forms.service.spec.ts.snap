// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsService createCustomDailyScheduleFormGroup should create form group 1`] = `
Object {
  "date": "",
  "from": "",
  "to": "",
}
`;

exports[`FormsService createLaneFormGroup should create form group 1`] = `
Object {
  "color": "#ffffff",
  "id": Any<String>,
  "name": "",
}
`;

exports[`FormsService createProductAvailabilityFormGroup should create form group 1`] = `
Object {
  "dayFrom": "",
  "dayTo": "",
  "price": "",
  "timeFrom": "",
  "timeTo": "",
  "type": "A",
}
`;

exports[`FormsService createProductConfigSetFormGroup should create form group 1`] = `
Object {
  "items": Array [],
  "position": 0,
  "productSetId": "",
}
`;

exports[`FormsService createProductConfigSetItemFormGroup should create form group 1`] = `
Object {
  "netPackagingFee": 0,
  "position": 0,
  "price": 0,
  "productComponentId": "",
  "refGroupPrice": 0,
}
`;

exports[`FormsService createProductVariantFormGroup should create form group 1`] = `
Object {
  "availabilities": Array [],
  "id": Any<String>,
  "isAvailable": true,
  "netPackagingFee": 0,
  "pack": Object {
    "size": 0,
    "unit": "",
  },
  "position": 0,
  "price": 0,
  "refGroupPrice": 0,
  "variantName": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
}
`;

exports[`FormsService createRatingPolicyFormGroup should create form group  1`] = `
Object {
  "description": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "key": "",
  "ratings": Array [],
  "title": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
}
`;

exports[`FormsService createRkeeperFormGroup should create form group with anyupp password 1`] = `
Object {
  "anyuppPassword": "",
  "anyuppUsername": "",
  "endpointUri": "",
  "rkeeperPassword": "",
  "rkeeperUsername": "",
}
`;

exports[`FormsService createRkeeperFormGroup should create form group without anyupp password 1`] = `
Object {
  "anyuppPassword": "",
  "anyuppUsername": "",
  "endpointUri": "",
  "rkeeperPassword": "",
  "rkeeperUsername": "",
}
`;

exports[`FormsService createTipPolicyFormGroup should create form group  1`] = `
Object {
  "description": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "maxOtherAmount": null,
  "percents": Array [],
  "title": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
}
`;
